### ISSUE TITLE
False positive `object, array, and null values should not be evaluated in template with ${{ }}` with lists in matrix

### ISSUE THREAD

**Issue by nantiferov** [rhysd/actionlint#77]
Hi,

I think I might found a bug related to feature added in [v1.6.2](https://github.com/rhysd/actionlint/blob/b713ef171f6d65460fe53a5b5d9fc9415856447d/CHANGELOG.md#v162---23-aug-2021), `actionlint now checks evaluated values at ${{ }} are not an object nor an array since they are not useful`

Example below produces this error, however it's a valid workflow and recommended for self-hosted runners (multiple tags to select runner).
https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#example-7

[Link to playground](https://rhysd.github.io/actionlint/#eJx9kLsOgkAQRXu+4ha0WBmL7WkpSKyMxQoTWQMD2YfBEP5dlpduYqwmc3LuzM6ybEjAkrFRywKXzpnqGj3amxERZu4roB2bxBvxMKCRVqv+MDEmjXGcDWO1tHR/LT5WaeuguKhdSXsPJFgGCHyYp1I3p2OIasWu/5UMohPvw2QQNZY6s+1PwPPlaVG1KMlKVZs9Sfz8eifyc5al+Z/T1w8SID8tXvw3KfBYpA==)

Error message:
`object, array, and null values should not be evaluated in template with ${{ }} but evaluating the value of type array<string> [expression]`

Example:
```yaml
name: test
on: [push]
jobs:
  test:
    runs-on: ${{ matrix.runner }}
    strategy:
      matrix:
       include:
         - runner: 
           - arm64
           - linux
         - runner:  
            - x64
            - linux
    steps:
      - name: Echo details
        env:
          RUNNER: ${{ matrix.runner }}
        run: echo $RUNNER
```

I tried to change it to `${{ toJSON(matrix.runner) }}`, actionlint passes after that, but workflow is broken after that.

**Comment by dekimsey** [comment]
I ran into this with another variation of this is that is also supported behavior by GHA:

```yaml
jobs:
  build:
    strategy:
      matrix:
        include:
          - { runner: [os=macos, x64]}
          - { runner: [os=macos-arm, arm64]}
      fail-fast: false
    runs-on: [self-hosted, "${{matrix.runner}}"]
    steps:
      ...
```

**Comment by rhysd** [comment]
actionlint looks correct to me. `matrix.runner` is an array, not a string. So it cannot be embedded with `${{ }}`. You need to put each elements

```yaml
- run: ${{ matrix.runner[0] }} ${{ matrix.runner[1] }}
```

or serialize it to JSON string

```yaml
- run: ${{ toJSON(matrix.runner) }}
```
